text_extraction.go

```diff
@@ -0,0 +1,124 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notion
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	notion "github.com/jomei/notionapi"
+	"github.com/tidwall/gjson"
+)
+
+type extractor func(notion.Block) (string, error)
+
+var errNoExtractor = errors.New("no extractor")
+
+var titleExtractor = extractor(func(block notion.Block) (string, error) {
+	bytes, err := json.Marshal(block)
+	if err != nil {
+		return "", fmt.Errorf("failed marshalling into JSON: %w", err)
+	}
+	return gjson.Get(string(bytes), block.GetType().String()+".title").Str, nil
+})
+
+var plainTextExtractor = extractor(func(block notion.Block) (string, error) {
+	bytes, err := json.Marshal(block)
+	if err != nil {
+		return "", fmt.Errorf("failed marshalling into JSON: %w", err)
+	}
```

Comment:
I think these lines deserve a function, especially with the redundancy of having the same error message in all of the extractors.

Reply:
oh makes sense! it's already a marshal error msg, so should be enough